### PR TITLE
The belt laid schemes of mice and men often go awry. (Belt tweak PR)

### PR DESCRIPTION
**Pull request by MrHorizons**
Blame the bad PR title on [pungenerator.org](https://pungenerator.org).

A semi-port of [BeeStation #1349](https://github.com/BeeStation/BeeStation-Hornet/pull/1349), with some elements adapted to fit the lack of plumbing in Yog. Unlike my other PRs, the changelog is fairly comprehensive.

#### Changelog

:cl:  
rscadd: RCDs can now deconstruct airlock assemblies and firelock frames.
rscadd: Biobags can carry organs, toolbelts can carry RCDs, RPDs, and inducers, and grenadier belts can carry combustible lemons and cherrybombs.
tweak: Medical belts and toolbelts can hold greater total weight.
tweak: RCDs and RPDs can be worn on the belt slot.
tweak: Liver, lungs, and stomach organs are small.
/:cl:

### CODE REVIEW

**Comment by Xoxeyos** [comment]
Sweet stuff dude, I needed this.

**Review by Amelia0010 — APPROVED** [review]

File: code/game/machinery/doors/firedoor.dm [hunk 1]
```diff
@@ -442,6 +442,19 @@
 	..()
 	icon_state = "frame[constructionStep]"
 
+/obj/structure/firelock_frame/rcd_vals(mob/user, obj/item/construction/rcd/the_rcd)
+	if(the_rcd.mode == RCD_DECONSTRUCT)
+		return list("mode" = RCD_DECONSTRUCT, "delay" = 50, "cost" = 16)
+	return FALSE
+
+/obj/structure/firelock_frame/rcd_act(mob/user, obj/item/construction/rcd/the_rcd, passed_mode)
+	switch(passed_mode)
```
Comment (Amelia0010) [inline code/game/machinery/doors/firedoor.dm]:
```suggestion
	if(passed_mode == RCD_DECONSTRUCT)
```

No need for a 1 item switch

File: code/game/objects/structures/door_assembly.dm [hunk 2]
```diff
@@ -312,3 +312,16 @@
...
 			new mineral_path(T, 2)
 	qdel(src)
+
+/obj/structure/door_assembly/rcd_vals(mob/user, obj/item/construction/rcd/the_rcd)
+	if(the_rcd.mode == RCD_DECONSTRUCT)
+		return list("mode" = RCD_DECONSTRUCT, "delay" = 50, "cost" = 16)
+	return FALSE
+
+/obj/structure/door_assembly/rcd_act(mob/user, obj/item/construction/rcd/the_rcd, passed_mode)
+	switch(passed_mode)
```
Comment (Amelia0010) [inline code/game/objects/structures/door_assembly.dm]:
```suggestion
	if(passed_mode == RCD_DECONSTRUCT)
```

**Review by Amelia0010 — CHANGES_REQUESTED** [review]
wrong button l m a o

File: code/game/machinery/doors/firedoor.dm (hunk 1 above)
Comment (Amelia0010) [reply code/game/machinery/doors/firedoor.dm]: Also make sure to change the line below it

File: code/game/objects/structures/door_assembly.dm (hunk 2 above)
Comment (Amelia0010) [reply code/game/objects/structures/door_assembly.dm]: also make sure to change the line below it

File: code/game/machinery/doors/firedoor.dm (hunk 1 above)
Comment (MrHorizons) [reply code/game/machinery/doors/firedoor.dm]: It appears that making the suggested change to firedoor.dm launches me down a rabbit hole of endless errors that don't seem to go away no matter what I do.

Comment (Amelia0010) [reply code/game/machinery/doors/firedoor.dm]: What exactly are you changing it to?
Line 452 should be struck, line 451 changed to the above change, and then line 453-455 should be de-indented by one indent

Comment (Amelia0010) [reply code/game/machinery/doors/firedoor.dm]: So lines 451 through 456 would become

```
	If(passed_mode == RCD_DECONSTRUCT)
		to_chat(user, "<span class='notice'>You deconstruct [src].</span>")
		qdel(src)
		return TRUE
	return FALSE

Comment (MrHorizons) [reply code/game/machinery/doors/firedoor.dm]: ... well, this is _absolutely_ spooky. I made those exact changes before and they didn't work, but now they do. Weird.

**Comment by JamieD1** [comment]
Thanks Nich

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**